energymeters/integrationpluginenergymeters.cpp

```diff
@@ -0,0 +1,394 @@
+*/
+    m_slaveIdParamTypeIds.insert(pro380ThingClassId, pro380ThingSlaveAddressParamTypeId);
+    m_slaveIdParamTypeIds.insert(sdm630ThingClassId, sdm630ThingSlaveAddressParamTypeId);
+
+    m_modbusUuidParamTypeIds.insert(pro380ThingClassId, pro380ThingModbusMasterUuidParamTypeId);
+    m_modbusUuidParamTypeIds.insert(sdm630ThingClassId, sdm630ThingModbusMasterUuidParamTypeId);
+
+    m_connectionStateTypeIds.insert(pro380ThingClassId, pro380ConnectedStateTypeId);
+    m_connectionStateTypeIds.insert(sdm630ThingClassId, sdm630ConnectedStateTypeId);
+
+    m_voltageL1StateTypeIds.insert(pro380ThingClassId, pro380VoltageL1StateTypeId);
+    m_voltageL1StateTypeIds.insert(sdm630ThingClassId, sdm630VoltageL1StateTypeId);
+
+    m_voltageL2StateTypeIds.insert(pro380ThingClassId, pro380VoltageL2StateTypeId);
+    m_voltageL2StateTypeIds.insert(sdm630ThingClassId, sdm630VoltageL2StateTypeId);
+
+    m_voltageL3StateTypeIds.insert(pro380ThingClassId, pro380VoltageL3StateTypeId);
+    m_voltageL3StateTypeIds.insert(sdm630ThingClassId, sdm630VoltageL3StateTypeId);
+
+    m_currentL1StateTypeIds.insert(pro380ThingClassId, pro380CurrentL1StateTypeId);
+    m_currentL1StateTypeIds.insert(sdm630ThingClassId, sdm630CurrentL1StateTypeId);
+
+    m_currentL2StateTypeIds.insert(pro380ThingClassId, pro380CurrentL2StateTypeId);
+    m_currentL2StateTypeIds.insert(sdm630ThingClassId, sdm630CurrentL2StateTypeId);
+
+    m_currentL3StateTypeIds.insert(pro380ThingClassId, pro380CurrentL3StateTypeId);
+    m_currentL3StateTypeIds.insert(sdm630ThingClassId, sdm630CurrentL3StateTypeId);
+
+    m_activePowerStateTypeIds.insert(pro380ThingClassId, pro380CurrentPowerEventTypeId);
+    m_activePowerStateTypeIds.insert(sdm630ThingClassId, sdm630CurrentPowerStateTypeId);
+
+    m_frequencyStateTypeIds.insert(pro380ThingClassId, pro380FrequencyStateTypeId);
+    m_frequencyStateTypeIds.insert(sdm630ThingClassId, sdm630FrequencyStateTypeId);
+
+    m_powerFactorStateTypeIds.insert(pro380ThingClassId, pro380PowerFactorStateTypeId);
+    m_powerFactorStateTypeIds.insert(sdm630ThingClassId, sdm630PowerFactorStateTypeId);
+
+    m_totalEnergyConsumedStateTypeIds.insert(pro380ThingClassId, pro380TotalEnergyConsumedEventTypeId);
+    m_totalEnergyConsumedStateTypeIds.insert(sdm630ThingClassId, sdm630TotalEnergyConsumedEventTypeId);
+
+    m_totalEnergyProducedStateTypeIds.insert(pro380ThingClassId, pro380TotalEnergyProducedStateTypeId);
+    m_totalEnergyProducedStateTypeIds.insert(sdm630ThingClassId, sdm630TotalEnergyProducedStateTypeId);
+
+    m_discoverySlaveAddressParamTypeIds.insert(pro380ThingClassId, pro380DiscoverySlaveAddressParamTypeId);
+    m_discoverySlaveAddressParamTypeIds.insert(sdm630ThingClassId, sdm630DiscoverySlaveAddressParamTypeId);
+
+    m_registerMaps.insert(pro380ThingClassId, pro380RegisterMap);
+    m_registerMaps.insert(sdm630ThingClassId, sdm630RegisterMap);
+
+    // FIXME leads to crash, probably an issue in the modbus rtu resource
```

Comment:
This needs to be moved into the plugin `init()`, since that method will be called when the core initialized all resources and the plugins are ready to be initialized. The ctor of the plugin itself will be run during the load process of the plugin. The hases can remain here.

Reply:
done